src/components/collection/Summary.js

```diff
@@ -54,12 +64,30 @@ function Summary({
   });
   const classes = useStyles();
   const { t } = useTranslation();
-  const { hooks } = useContext(QueryClientContext);
+  const { hooks, useMutation } = useContext(QueryClientContext);
   const { data: categories } = hooks.useItemCategories(itemId);
   const { data: allCategories } = hooks.useCategories();
   const categoriesDisplayed = allCategories?.filter((category) =>
     categories?.map((entry) => entry.categoryId).includes(category.id),
   );
+  const { mutate: postFlagItem } = useMutation(MUTATION_KEYS.POST_ITEM_FLAG);
+  const [open, setOpen] = useState(false);
+  const [selectedFlag, setSelectedFlag] = useState(false);
+
+  const { data: flags } = hooks.useFlags();
+
+  const openItemFlagMenu = () => {
+    setOpen(true);
+  }
+
+  const onFlag = () => {
+    postFlagItem({
+      flagId: selectedFlag.id,
+      itemId,
+    });
+    setOpen(false);
+  };
+
```

Comment:
Do you think we can create a whole component `FlagItemButton`? And import it from graasp-ui?

Reply:
Probably the button will also contain the dialog (so both stay together).

Reply:
I tried to add an additional component in graasp-ui, however it's not functioning well. When the dialog is opened, no matter where I click, it will also trigger a click on the flagItemButton, (so it will always trigger an 'open dialog' action, when user click anywhere when the dialog is open, even if the mouse-pointer is not on the dialog window). As a result the dialog can't be closed once it's open.
So I created a component for the FlagItemButton in explore instead. I use the same piece of code and it's working well. Please let me know if you have any idea, I'd love to add it to graasp-ui.